packages/react/src/components/ContentSwitcher/ContentSwitcher.js

```diff
@@ -63,18 +77,43 @@ export default class ContentSwitcher extends React.Component {
         ]),
         onKeyDown: this.handleChildChange,
         selected: index === this.state.selectedIndex,
+        ref: switchRefs[index],
       })
     );
   }
 
   handleChildChange = data => {
+    // the currently selected child index
     const { selectedIndex } = this.state;
+    // the newly selected child index
     const { index } = data;
+    const { key } = data;
 
-    if (selectedIndex !== index) {
-      this.setState({ selectedIndex: index });
-      this.props.onChange(data);
+    if (key) {
```

Comment:
Would it make sense for the key checking logic to live here? In other words, moving the `matches` block for arrow keys to this block. Going through this, it wasn't immediately clear that we should only be running this branch with `getNextIndex` IFF the event key is an arrow key.

Then, on the child side, we can bubble up the event as they come in.

Reply:
is there maybe a way to clean this up without further refactoring that would be satisfactory? I do agree having that logic in the Switch component is a little less intuitive than it could be.

Reply:
I guess duplicating the logic that checks for the right keys?

Reply:
Well, that doesn't sound ideal. I'll make the suggested change then 👍